docs/features/deserializing-events.md

```diff
@@ -4,10 +4,35 @@ layout: default
 ---
 
 ## Deserializing Events
-We provide support for deserializing events to typed objects:
+
+We provide support for deserializing events to typed event objects where the custom event payload is available via the `.GetPayload()` method.
+The original event payload can also be accessed via the `.Data` property.
 
 ```csharp
-var eventGridMessage = EventGridParser.Parse<BlobCreated>(rawEvent);
+// Parse from your custom event implementation with the `.Parse<>` function.
+EventGridBatch<NewCarRegistered> eventGridBatch = EventGridParser.Parse<NewCarRegistered>(rawEvent);
+
+// The event data type will be wrapped inside an `EventGridEvent<>` instance.
+NewCarRegistered eventGridMessage = eventGridBatch.Events.First();
+
+// The original event payload can now be accessed.
+CarEventData typedEventPayload = eventGridMessage.GetPayload();
+object untypedEventPaylaod = eventGridMessage.Data;
+```
+
+Or from event data type objects where the event data payload is available via the `.GetPayload()` method.
+The original event payload can also be accessed via the `.Data` property.
+
+```csharp
```

Comment:
Given only the first line is different we might want to ditch the rest of the sample as it's less relevant?

Reply:
Yeah, ok.

Reply:
Hmm, not sure if I agree, it helps with describing the `.GetPayload()` method. Removing these would make it IMO so stripped that I wouldn't be sure about what this doc file should talk about.

Reply:
I would only do it for the bottom sample given it's 99% copy and only increases maintainability but ok

Reply:
Yep. Done.